src/main/java/com/agiletec/plugins/jacms/apsadmin/content/attribute/manager/ResourceAttributeManager.java

```diff
@@ -40,26 +40,31 @@ protected void updateAttribute(AttributeInterface attribute, AttributeTracer tra
         super.updateAttribute(attribute, tracer, request);
         IResourceManager resourceManager = (IResourceManager) ApsWebApplicationUtils.getBean(JacmsSystemConstants.RESOURCE_MANAGER, request);
         AbstractResourceAttribute resourceAttribute = (AbstractResourceAttribute) attribute;
-        if (null != resourceAttribute.getMetadatas()) {
-            resourceAttribute.getMetadatas().clear();
-        }
+
         Map<String, List<String>> mapping = resourceManager.getMetadataMapping();
         if (null == mapping || mapping.isEmpty()) {
             return;
         }
+
         List<Lang> langs = this.getLangManager().getLangs();
         for (Lang currentLang : langs) {
             tracer.setLang(currentLang);
             String formFieldPrefix = tracer.getFormFieldName(attribute) + "_metadata_";
             mapping.keySet().stream().forEach(key -> {
                 String value = request.getParameter(formFieldPrefix + key);
-                if (!StringUtils.isBlank(value)) {
+                if ( value != null) {
+                    // all this update is heavily based on metadata presents in the http request.
+                    // So I will skip update metadata if a particular key is not present inside the current http request.
+                    // solution proposed by Eugenio
+                    if (value.trim().length() == 0) {
+                        value = "";
+                    }
```

Comment:
@eugeniosant Setting the value to `null` triggers other problems. Are you really sure the only problem is inside `addResourceMetadatasElement` ? ... you found that by chance, maybe there are others, for me the more conservative solution is set to empty string not to `null` (less probability to have null pointer).
I mentioned you because is your solution I just cherry-picked the code you send to me, for me the solution should be different: maybe to change the code inside the `addList` function and remove the resource conditionally the upgradeResource inside the `choose` function if it is really used from others.